app/data_model/questionnaire_store.py

```diff
@@ -35,14 +33,14 @@ def save(self):
         logger.debug("Saving user data %s for user id %s", self.data, self.user_id)
         self.storage.store(data=self.data, user_id=self.user_id, user_ik=self.user_ik)
 
-    def encode_metadata(self, metadata):
-        self.data["METADATA"] = jsonpickle.encode(metadata)
-
-    def decode_metadata(self):
+    @property
+    def metadata(self):
         if "METADATA" in self.data:
-            return jsonpickle.decode(self.data["METADATA"])
-        else:
-            raise RuntimeError("No metadata for user %s", self.user_id)
```

Comment:
Both these methods are *only* each accessed once as part of the authenticator calls. 

The set method is called once metadata has been validated through parse_metadata, so couldn't ever be anything than a valid metadata dict, meaning we can't save anything other than valid metadata.

The call to get_metadata is the main accessor of the metadata property on the questionnaire_store and it returns None when hitting an anonymous user.